string data
bool shouldCheckForID
uint8[] agentIDs